src/extensions/default/bramble/lib/LinkManagerRemote.js

```diff
@@ -13,8 +16,19 @@
             return true;
         }
 
-        if(!(/\:?\/\//.test(url)) && window._Brackets_LiveDev_Transport) {
+        var pathNav = !(/\:?\/\//.test(url));
+        // `internalNav` handles the special case of fragment ids in the
+        // same html page in preview mode (not tutorial mode)
+        var internalNav = /^#.*/.test(href);
+
+        if(pathNav && window._Brackets_LiveDev_Transport) {
             window._Brackets_LiveDev_Transport.send("bramble-navigate:" + url);
+        } else if(internalNav) {
+            element = document.querySelector(href) || document.getElementsByName(href.slice(1));
```

Comment:
Based on parsing rules for frag-ids, it first looks for a matching `id` and if that is not found, it looks for a matching `name` attribute